src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,387 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import Utils from '../../../../Utils';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import { User } from '@microsoft/microsoft-graph-types';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  planId?: string;
+  planName?: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+  bucketId?: string;
+  bucketName?: string;
+  title: string;
+  startDateTime?: string;
+  dueDateTime?: string;
+  percentComplete?: number;
+  assignedToUserIds?: string;
+  assignedToUserNames?: string;
+  description?: string;
+  orderHint?: string;
+}
+
+class PlannerTaskAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner Task';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    telemetryProps.bucketId = typeof args.options.bucketId !== 'undefined';
+    telemetryProps.bucketName = typeof args.options.bucketName !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.startDateTime = typeof args.options.startDateTime !== 'undefined';
+    telemetryProps.dueDateTime = typeof args.options.dueDateTime !== 'undefined';
+    telemetryProps.percentComplete = typeof args.options.percentComplete !== 'undefined';
+    telemetryProps.assignedToUserIds = typeof args.options.assignedToUserIds !== 'undefined';
+    telemetryProps.assignedToUserNames = typeof args.options.assignedToUserNames !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.orderHint = typeof args.options.orderHint !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['title', 'planId', 'bucketId'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getPlanId(args)
+      .then((planId: string) => 
+        this.getBucketId(args, planId).then(async (bucketId: string) => {
```

Comment:
Let's not nest promises because it makes the code harder to read. Instead, let's chain them.